operations-stdlib/src/commonMain/kotlin/string/CompareToDate.kt

```diff
@@ -0,0 +1,126 @@
+        onFailure = { null }
+    )
+
+    private fun String.splitDate(): List<String> { return this.split(DELIMITER) }
+    private fun List<String>.toDateParameters(): DateParameters? = runCatching {
+        DateParameters(
+            year = get(YEAR_INDEX).toInt(),
+            month = get(MONTH_INDEX).toInt(),
+            day = get(DAY_INDEX).toInt()
+        )
+    }.fold(
+        onSuccess = { it },
+        onFailure = { null }
+    )
+
+    private fun List<String>.isContainsValidDate(): Boolean = runCatching {
+        val month = get(MONTH_INDEX)
+        val day = get(DAY_INDEX)
+        return get(YEAR_INDEX).length <= YEAR_LENGTH
+            && month.length == MONTH_OR_DAY_LENGTH
+            && month.toInt() <= MONTH_MAX_VALUE
+            && month.toInt() >= MONTH_MIN_VALUE
+            && day.length == MONTH_OR_DAY_LENGTH
+            && day.toInt() <= DAY_MAX_VALUE
+            && day.toInt() >= DAY_MIN_VALUE
+    }.fold(
+        onSuccess = { it },
+        onFailure = { false }
+    )
+
+    private fun CompareDateArguments.invokeCompare(): Int? = runCatching {
+        val yearComparationStatus = this.compareYears() ?: return null
+        val monthComparationStatus = this.compareMonth() ?: return null
+        val dayComparationStatus = this.compareDay() ?: return null
+
+        return if (yearComparationStatus != DateComparationStatus.EQUALS) {
+            yearComparationStatus.raw
+        } else if (monthComparationStatus != DateComparationStatus.EQUALS) {
+            monthComparationStatus.raw
+        } else {
+            dayComparationStatus.raw
+        }
+    }.fold(
+        onSuccess = { it },
+        onFailure = { null }
+    )
+
+    private fun CompareDateArguments.compareYears(): DateComparationStatus? = when {
+        baseDate.year == compareDate.year -> DateComparationStatus.EQUALS
+        baseDate.year > compareDate.year -> DateComparationStatus.BEFORE
+        baseDate.year < compareDate.year -> DateComparationStatus.AFTER
+        else -> null
+    }
+
+    private fun CompareDateArguments.compareMonth(): DateComparationStatus? = when {
+        baseDate.month == compareDate.month -> DateComparationStatus.EQUALS
+        baseDate.month > compareDate.month -> DateComparationStatus.BEFORE
+        baseDate.month < compareDate.month -> DateComparationStatus.AFTER
+        else -> null
+    }
+
+    private fun CompareDateArguments.compareDay(): DateComparationStatus? = when {
+        baseDate.day == compareDate.day -> DateComparationStatus.EQUALS
+        baseDate.day > compareDate.day -> DateComparationStatus.BEFORE
+        baseDate.day < compareDate.day -> DateComparationStatus.AFTER
+        else -> null
+    }
+}
+
+private data class CompareDateArguments(
+    val baseDate: DateParameters,
+    val compareDate: DateParameters
+)
+private data class DateParameters(
+    val year: Int,
+    val month: Int,
+    val day: Int
+
+)
+enum class DateComparationStatus(val raw: Int) {
```

Comment:
wdyt?
```suggestion
enum class DateComparisonStatus(val raw: Int) {
```

Reply:
I think I can remove this enum like @annaszachiewicz suggested. WDYT?